README.md

```diff
@@ -1,3 +1,171 @@
+  a binary that can be executed by it self, aka, does not rely on a TensorFlow
+  or PyTorch runtime. In this mode all ops must be supported. 
+
+### Numbers of Typical Workloads
+
+By evaluating BladeDISC using a set of typical machine learning workloads for
+production purpose, DISC shows up to 3x speedup compared with
+TensorFlow/PyTorch.
+
+![Numbers](./docs/pics/numbers.png)
+
+#### Advantage in Dynamic Shape Workloads
+
+Specifically, for the BERT large inference on T4 we provide in the
+[examples](./docs/tutorials/tensorflow_inference_and_training.md), static compiler
+optimization (XLA) shows severe performance degradation due to its compilation
+overhead, while DISC shows a 1.75x speedup.
+
+| TensorFlow  |    XLA    |    DISC    |
+|-------------|-----------|------------|
+|   1.78 s    |   41.69s  |    1.02s   |
+|   1X        |           |    1.75X   |
+
+## API QuickView
+
+### For TensorFlow Users
+
+Only two lines of code are needed on native Tensorflow program as the following:
+
+``` python
+import numpy as np
+import tensorflow as tf
+
+## enable BladeDISC on TensorFlow program
+import tensorflow_blade_disc as disc
+disc.enable()
+
+## construct TensorFlow Graph and run it
+g = tf.Graph()
+with g.as_default():
+    ...
+    with tf.session as sess:
+        sess.run(...)
+```
+
+For more information, please refer to [QuickStart for TensorFlow
+Users](./docs/quickstart.md#quickstart-for-tensorflow-users)
+
+## For PyTorch Users
+
+PyTorch users only need the following few lines of code to enable
+BladeDISC:
+
+``` python
+import torch_blade
+# construct PyTorch Module
+class MyModule(nn.Module):
+    ...
+
+module = MyModule()
+
+with torch.no_grad():
+    # blade_module is the optimized module by BladeDISC
+    blade_module = torch_blade.optimize(module, allow_tracing=True, model_inputs=(x, y))
+
+# run the optimized module
+blade_module(x, y)
+```
+
+`torch_blade.optimize` accepts an `nn.Module` object and outputs the
+optimized module.  For more information, please refer to [Quickstart
+for PyTorch Users](./docs/quickstart.md#quickstart-for-pytorch-users).
+
+## Setup and Examples
+
+* [How to setup and build from source](./docs/build_from_source.md)
+* [Use case of TensorFlow Inference and Training](./docs/tutorials/tensorflow_inference_and_training.md)
+* [Use case of PyTorch Inference](./docs/tutorials/torch_bert_inference.md)
+
+## Publications
+
+* [DISC: A Dynamic Shape Compiler for Machine Learning
+  Workloads](https://arxiv.org/pdf/2103.05288.pdf)
+
+## Tutorials and Documents for Developers
+
+* [Tutorial: A Walkthough of the BladeDISC Pass Pipeline](./docs/developers/pass_pipeline.md)
+* [Introduction on Runtime Abstraction Layer](./docs/developers/runtime_abstraction_layer.md)
+* [BladeDISC Torch Overview](./docs/developers/bladedisc_torch_overview.md)
+* [Tutorial: How to add the support of a new Torch Op](./docs/developers/torch_add_a_new_converter.md)
```

Comment:
How To Add a New Torch Operator Converter

Reply:
done